src/stores/LayoutStore.ts

```diff
@@ -185,6 +195,64 @@ export class LayoutStore {
         }
     };
 
+    @flow.bound *renameLayout(oldName: string, newName: string) {
+        const appStore = AppStore.Instance;
+
+        if (!this.layouts || !newName || !this.dockedLayout) {
+            appStore.alertStore.showAlert("Save layout failed! Empty layouts or name.");
+            return;
+        }
+
+        if (appStore.layoutStore.layoutExists(newName)) {
+            appStore.alertStore.showAlert("Layout name already exists.");
+            return;
+        }
+
+        if (!oldName || !this.layoutExists(oldName)) {
+            appStore.alertStore.showAlert(`Cannot rename layout ${oldName}! It does not exist.`);
+            return;
+        }
+
+        appStore.dialogStore.hideSaveLayoutDialog();
+
+        // save layout to layouts[] & server/local storage
+        const configToSave = this.layouts[oldName];
+        this.layouts[newName] = configToSave;
+        if (!PresetLayout.isPreset(this.layoutNameToBeSaved)) {
+            try {
+                const success = yield appStore.apiService.setLayout(newName, configToSave);
+                if (success) {
+                    appStore.apiService.clearLayout(oldName).then(
```

Comment:
Could we use `yield` here? Maybe also adjust `deleteLayout` to use `@flow.bound` and `yield`.

Reply:
@TienHao Maybe we can remove `then()` by
```
try {
    const success = yield appStore.apiService.setLayout(newName, configToSave);
    if (success) {
        const success = yield appStore.apiService.clearLayout(oldName);
        if (success) {
            delete this.layouts[oldName];
        }
        this.handleRenameResult(oldName, newName, success);
    }
} catch (err) {
    console.log(err);
    this.handleRenameResult(oldName, newName, false);
}
```
We can also try to remove `then()` in `deleteLayout` similarly.